src/modules/discordInvitesMonitor.module.ts

```diff
@@ -0,0 +1,135 @@
+import Module from "./module.js";
+import { actualMention, isSpecialUser } from "../util/users.js";
+import { config } from "../Config.js";
+import { getMember } from "../util/member.js";
+import { GuildMember } from "discord.js";
+import { getOrCreateUserById } from "../store/models/DDUser.js";
+import { getTierByLevel } from "./xp/xpForMessage.util.js";
+
+const invitePatterns = [
+  /discord\.gg\/[a-zA-Z0-9]+/gi,
+  /discordapp\.com\/invite\/[a-zA-Z0-9]+/gi,
+  /discord\.com\/invite\/[a-zA-Z0-9]+/gi,
+  /(^| )\.gg\/[a-zA-Z0-9]+/gi,
+];
+
+const whitelistDomains: string[] = []; // For any .gg domains that are not discord.gg
```

Comment:
while a nice consideration, this is functionally useless at the moment, as it's not persistent and can't be edited in any way